openms_python/py_aasequence.py

```
@@ -204,26 +218,126 @@ def __eq__(self, other: object) -> bool:
-        return residue.getOneLetterCode()
+        if isinstance(index, slice):
+            start, stop, step = index.indices(len(self))
+            if step != 1:
+                raise ValueError("Step slicing is not supported for amino acid sequences")
+            return Py_AASequence.from_native(self._sequence.getSubsequence(start, stop - start))
+        else:
+            # Handle negative indices
+            if index < 0:
+                index = len(self) + index
+            if index >= len(self):
+                raise IndexError(f"Index {index} out of range for sequence of length {len(self)}")
```

Ruff (0.14.7)

Avoid specifying long messages outside the exception class

(TRY003)